app/src/battery.c

```diff
@@ -35,7 +38,16 @@ static int zmk_battery_update(struct device *battery) {
 
     LOG_DBG("Setting BAS GATT battery level to %d.", state_of_charge.val1);
 
-    return bt_gatt_bas_set_battery_level(state_of_charge.val1);
+    rc = bt_gatt_bas_set_battery_level(state_of_charge.val1);
```

Comment:
Should there be an event listener in the bluetooth files that handles this change?

Reply:
That's debatable. Until we have some other true "source of truth for the battery level", other than the BAS service, I think this makes sense, since this is recording the updated state.

I see a future refactor for `app/src/battery.c` to keep the state locally, then we pull the updating of the BAS GATT service into an event handler.